fix(otlptrace,otlpmetric): remove endpoint URL path cleaning

mjq opened the pull request:
When setting an explicit OTLP traces endpoint URL (via `otlptracehttp.WithEndpointURL` or the `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` environment variable), any given trailing slash is stripped. This makes it impossible to export traces to an endpoint requiring a trailing slash. It also conflicts with [the spec](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#endpoint-urls-for-otlphttp):

> For the per-signal variables (OTEL_EXPORTER_OTLP_\<signal\>_ENDPOINT), the URL MUST be used as-is without any modification. The only exception is that if an URL contains no path part, the root path / MUST be used (see [Example 2](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#example-2)).

This stripping happens due to [the use of `path.Clean` in `otlpconfig.cleanPath`](https://github.com/open-telemetry/opentelemetry-go/blob/b4b461d050f1d0f5bda89ddac718180162120606/exporters/otlp/otlptrace/otlptracehttp/internal/otlpconfig/options.go#L97). From [the `path.Clean` docs](https://pkg.go.dev/path#Clean):

> The returned path ends in a slash only if it is the root "/".

Other signals appear to use `path.Clean` and might therefore have the same bug, but I've only tested and fixed traces.

Fixes #6709.

dmathieu commented:
> the URL MUST be used as-is without any modification

This kind of indicates we should remove `path.Clean` entirely.

pellared commented:
@mjq, ping :wink:


See: https://github.com/open-telemetry/opentelemetry-go/pull/6710#issuecomment-2837806161

mjq commented:
@pellared Sorry for the delay! I incorporated @dmathieu's feedback re: removing the usage of `path.Clean`. I removed it from the equivalent config option in metrics as well for consistency.

I also realized I was changing generated files in the original commit - now the changes are made to the templates and applied with `make generate`.

mjq commented:
I fixed the failing changelog check (bad automatic merge with `main`).

> What about otlplog?

@pellared With this change `path.Clean` no longer appears in the codebase, and from poking around I couldn't find any equivalent code for log configs.

pellared commented on CHANGELOG.md:
```diff
@@ -33,6 +33,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
+### Fixed
+
+- Stop stripping trailing slashes from configured endpoint URLs. (#6710)
```
```suggestion
- Stop stripping trailing slashes from configured endpoint URLs in `go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc`. (#6710)
- Stop stripping trailing slashes from configured endpoint URLs in `go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp`. (#6710)
- Stop stripping trailing slashes from configured endpoint URLs in `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc`. (#6710)
- Stop stripping trailing slashes from configured endpoint URLs in `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp`. (#6710)
```

mjq replied:
@pellared Committed these changes - thanks!

pellared commented:
I noticed that `otlplog` does not even do `strings.TrimSpace` :wink:

mjq commented:
@pellared @dmathieu I don't have permission to merge this, so please go ahead if everything looks good or let me know if there are any further changes I should make 👍 Thanks!

pellared commented:
@mjq, thank you very much for your contribution.
